### PR TITLE
Cleaning

### PR DESCRIPTION
**Pull request by BeaudanBrown**
Heap of linting, eslint warning/error removal, and fixed a couple small bugs found along the way

### CODE REVIEW

File: js/models/conversations.js [hunk 1]
```diff
@@ -239,37 +238,39 @@
...
-        if (message.isFriendRequest() && message.attributes.friendStatus === 'pending') return true;
-      }
-
-      return false;
+      const pendingFriendRequest =
+        messages.models.find(message =>
+          message.isFriendRequest() &&
+          message.attributes.friendStatus === 'pending'
+        );
+      return pendingFriendRequest !== undefined;
```
Comment (Mikunj) [inline js/models/conversations.js]: Would a `!!pendingFriendRequest` suffice here?

File: js/models/conversations.js [hunk 2]
```diff
@@ -738,17 +741,16 @@
...
       if (_options.direction === 'incoming') {
         const requests = await this.getPendingFriendRequests('incoming');
-        
-        for (const request of requests) {
-          // Delete the old message if it's pending
-          await this._removeMessage(request.id);
-        }
+
+        // Delete the old message if it's pending
+        await Promise.all(requests.map(async request => this._removeMessage(request.id)));
```
Comment (Mikunj) [inline js/models/conversations.js]: is `async` here needed?

File: js/models/conversations.js [hunk 3]
```diff
@@ -2115,29 +2122,34 @@
...
+      });
+      Whisper.Notifications.add({
+        conversationId: conversation.id,
+        iconUrl,
+        isExpiringMessage: false,
+        message: i18n(message, conversation.getTitle()),
+        messageSentAt: Date.now(),
+        title: i18n(title),
+      });
+      return Promise.resolve();
```
Comment (Mikunj) [inline js/models/conversations.js]: We can remove `return Promise.resolve()` as `async` functions automatically resolve when they finish executing

File: js/models/conversations.js [hunk 4]
```diff
@@ -1024,26 +1026,31 @@
...
           const outgoingRequests = await this.getPendingFriendRequests('outgoing');
           if (outgoingRequests.length > 0) {
-            // Check if the requests have errored, if so then remove them and send the new request if possible
-            const friendRequestSent = false;
-            for (const outgoing of outgoingRequests) {
+            // Check if the requests have errored, if so then remove them
+            // and send the new request if possible
+            let friendRequestSent = false;
+            const promises = [];
+            outgoingRequests.forEach(async outgoing => {
```
Comment (BeaudanBrown) [inline js/models/conversations.js]:
```suggestion
            outgoingRequests.forEach(outgoing => {
```

File: libtextsecure/message_receiver.js [hunk 5]
```diff
@@ -251,7 +254,7 @@ MessageReceiver.prototype.extend({
       this.calledClose
     );
     // TODO: handle properly
-    return;
+    return Promise.resolve();
```
Comment (Mikunj) [inline libtextsecure/message_receiver.js]: Same as above

File: libtextsecure/message_receiver.js [hunk 6]
```diff
@@ -1040,7 +1047,7 @@ MessageReceiver.prototype.extend({
       }
 
       // Exit early since the friend request reply will be a regular empty message
-      return;
+      return Promise.resolve();
```
Comment (Mikunj) [inline libtextsecure/message_receiver.js]: Same as above

File: js/models/conversations.js (hunk 1 above)
Comment (BeaudanBrown) [inline js/models/conversations.js]:
```suggestion
      return !!pendingFriendRequest;
```

File: js/models/conversations.js (hunk 2 above)
Comment (BeaudanBrown) [inline js/models/conversations.js]:
```suggestion
        await Promise.all(requests.map(request => this._removeMessage(request.id)));
```

File: js/models/conversations.js [hunk 7]
```diff
@@ -2104,8 +2110,9 @@
     // Notification for friend request received
     async notifyFriendRequest(source, type) {
       // Data validation
-      if (!source) return Promise.reject('Invalid source');
-      if (!['accepted', 'requested'].includes(type)) return Promise.reject('Type must be accepted or requested.');
+      if (!source) return Promise.reject(new Error('Invalid source'));
+      if (!['accepted', 'requested'].includes(type))
+        return Promise.reject(new Error('Type must be accepted or requested.'));
```
[the remark this answers is not in the document]
Comment (Mikunj) [reply js/models/conversations.js]: Yep

File: js/models/conversations.js (hunk 3 above)
Comment (BeaudanBrown) [reply js/models/conversations.js]: The linter doesn't like there being no return value for async functions, can either put the flag to ignore this warning or just leave the redundant return

Comment (Mikunj) [reply js/models/conversations.js]: Is the rue being violated the `consistent-return`? If so then changing `return Promise.reject(...)` to `throw ...` should fix it

File: libtextsecure/message_receiver.js (hunk 6 above)
Comment (Mikunj) [reply libtextsecure/message_receiver.js]: you can `return null` i think to fix the consistent return. @sachaaaaa thoughts?

File: js/models/conversations.js (hunk 3 above)
Comment (BeaudanBrown) [reply js/models/conversations.js]: Aah yes I thought this wasn't the case but I was missing the inline return on the first line

File: js/models/conversations.js (hunk 1 above)
Comment (BeaudanBrown) [reply js/models/conversations.js]: I think I like leaving it as is actually, easier to read imo

File: libtextsecure/message_receiver.js (hunk 6 above)
Comment (sachaaaaa) [reply libtextsecure/message_receiver.js]: Yeah `return null` is fine !

File: js/models/conversations.js (hunk 1 above)
Comment (sachaaaaa) [reply js/models/conversations.js]: I have no strong opinion, as I'm not overly familiar with the `!!` notation.
So I'd say I leave it to the committer's preference.

**Pull request merged**